File: Cpp/880_Decoded_String_at_Index.cpp
class Solution {
public:
    string decodeAtIndex(string S, int K) {
        string decode = "";
        for(int i=0; i<S.size(); ++i){
            if(S[i]>='0' && S[i]<='9'){
                string temp = decode;
                for(int j=0; j<S[i]-'0'-1; ++j) decode += temp;
            } else {
                decode += S[i];
            }
            
            if(decode.size() >= K) return {decode[K-1]};
        }
        
        return "";
    }
};

File: Kotlin/880_Decoded_String_at_Index.kt
class Solution {
    fun decodeAtIndex(s: String, k: Int): String {
        var i = 0L
        
        s.forEach{ch ->
            if(ch >= '0' && ch <= '9') {
                if(i * (ch - '0') >= k) return decodeAtIndex(s, ((k - 1) % i).toInt() + 1)
                i *= ch - '0'
            } else if (++i == k.toLong()){
                return ch.toString()
            }
        }
            
        return ""
    }
}